### PR TITLE
fix failing completion at offset 0

### LINKED ISSUE #766: Failing completion at offset 0
Try to complete a random file at offset 0:
```
org.eclipse.jface.text.BadLocationException
	at org.eclipse.jface.text.AbstractDocument.get(AbstractDocument.java:967)
	at org.eclipse.core.internal.filebuffers.SynchronizableDocument.get(SynchronizableDocument.java:130)
	at org.eclipse.lsp4e.LSPEclipseUtils.toCompletionParams(LSPEclipseUtils.java:229)
	at org.eclipse.lsp4e.operations.completion.LSContentAssistProcessor.computeCompletionProposals(LSContentAssistProcessor.java:107)
	at org.eclipse.lsp4e.test.completion.CompleteCompletionTest.lambda$6(CompleteCompletionTest.java:529)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```
This is a consequence of #747

### PR DESCRIPTION
**Pull request by ghentschke**
fixes #766

### CODE REVIEW

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java [hunk 1]
```diff
@@ -226,7 +226,8 @@ public static CompletionParams toCompletionParams(URI fileUri, int offset, IDocu
 		Position start = toPosition(offset, document);
 		final var param = new CompletionParams();
 		try {
-			String positionCharacter = document.get(offset-1, 1);
+			int positionCharacterOffset = offset > 0 ? offset-1 : offset;
+			String positionCharacter = document.get(positionCharacterOffset, 1);
```
Comment (mickaelistria) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java:230]: It's most likely failing if document is empty.

Comment (ghentschke) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java:230]: The `BadLocationException` in the stack trace has been triggered by LSPEclipseUtils.java:229. The document.get method fails if the offset is <0. 
What is the expected behavior for `org.eclipse.lsp4e.LSPEclipseUtils.toCompletionParams(URI, int, IDocument, char[])` in this case? I assumed 0 would be approriate for the offset value.

Comment (ghentschke) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java:230]: Which test case fails here?

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java [hunk 2]
```diff
@@ -225,18 +225,20 @@ public static CompletionParams toCompletionParams(URI fileUri, int offset, IDocu
...
-			if (Chars.contains(completionTriggerChars, positionCharacter.charAt(0))) {
-				param.setContext(new CompletionContext(CompletionTriggerKind.TriggerCharacter, positionCharacter));
-			} else {
-				// According to LSP 3.17 specification: the triggerCharacter in CompletionContext is undefined if
-				// triggerKind != CompletionTriggerKind.TriggerCharacter
-				param.setContext(new CompletionContext(CompletionTriggerKind.Invoked, positionCharacter));
+		if (document.getLength() > 0) {
+			try {
+				int positionCharacterOffset = offset > 0 ? offset-1 : offset;
+				String positionCharacter = document.get(positionCharacterOffset, 1);
```
Comment (mickaelistria) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java:231]: Wouldn't this trigger a BadLocationException for empty document?

Comment (ghentschke) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java:231]: Covered by unit test. For an empty document this code would not be reached.

**Review by mickaelistria — CHANGES_REQUESTED** [review]
Thank you!

**Pull request merged**